### DISCUSSION TITLE
Is it possible to select and copy multi-line strings from the results

### DISCUSSION THREAD

**Discussion by martingbrown** [microsoft/azuredatastudio#22832] [Q&A] [answered]
I've an application that writes .Net exception stack traces to a database table. I often want to copy them so that I can paste them into Visual Studio's Exception explorer window. However this doesn't work because Azure Data Studio is removing the CR/LF characters from the strings before displaying them in the Results grid.

I can get around this by selecting with a replace and then I can replace the \r\n characters in a text editor with the correct characters. Something like this:

```
SELECT
    REPLACE(REPLACE(js.LastException, CHAR(13), '\r'), CHAR(10), '\n'),
FROM ...
```

Is there a better way of doing this though?

**Comment by erinstellato-ms** [comment]
Hey @martingbrown - when you ask if there is a better way of doing this....are you looking for a different approach in terms of the REPLACE code you're using here, or something else?  Just want to make sure I understand the question.  Thanks!

**Reply by martingbrown** [reply]
I'd really like the output in the results grid to include newline characters so I don't need to do anything odd in my select clause. Some other database clients I've used over the years allowed grid cells to be opened in a large text box by double clicking, thus allowing all the text to be read.

**Comment by erinstellato-ms** [comment, marked as answer]
@martingbrown ah, ok, there is a setting to disable that.  Use CRTL/CMD + ,  to get to Settings, and then search for "copy remove" and you should find the Results: Copy Remove New Line option.  Uncheck that and when you copy the text, the CR/LF should be preserved.

**Comment by martingbrown** [comment]
Thank you. That's mostly what I was after. I don't suppose there is a way to easily toggle that for a single copy operation? Changing the setting feels a bit permanent.

On the context menu, for instance, there is already "Copy", "Copy With Headers" and "Copy Headers". I guess this threw me off the idea of looking at settings, because I assumed there would be a fourth menu option "Copy with line endings" if this could be done.

**Comment by erinstellato-ms** [comment]
@martingbrown There is not a quicker way to toggle, or toggle selectively.  It's a setting at the app level.
